blessed/sequences.py

```diff
@@ -816,3 +891,121 @@ def horizontal_distance(ucs, term):
+            terminal sequence, or None if not a terminal sequence
+        - ``params``: a tuple of str parameters in the terminal sequence,
+            or None if not a terminal sequence
+
+    """
+    # simple terminal control characters,
+    ctrl_seqs = u'\a\b\r\n\x0e\x0f'
+
+    if any([ucs.startswith(_ch) for _ch in ctrl_seqs]):
+        return TextPart(ucs[:1], True, '?', None)
+
+    matching_seq = term and (
+        term._re_will_move.match(ucs) or
+        term._re_wont_move.match(ucs)
+    )
+    if matching_seq:
+        (identifier, ) = (k for k, v in matching_seq.groupdict().items()
+                          if v is not None)
+        name = identifier
+        params = term._param_extractors[identifier].match(ucs).groups()
+        return TextPart(matching_seq.group(), True,
+                        name, params if params else None)
+
+    # known multibyte sequences,
+    matching_seq = term and (
+        term._re_cub and term._re_cub.match(ucs) or
+        term._re_cuf and term._re_cuf.match(ucs)
+    )
+
+    if matching_seq:
+        return TextPart(matching_seq.group(), True, '?', None)
+
+    raise ValueError("identify_part called on nonsequence "
+                     "{!r}".format(ucs))
+
+
+def enumerate_by_position(parts):
+    """Iterate over TextParts with an index, subdividing printable strings.
+
+    The index is the length of characters preceding that TextPart, in other
+    words the cumulative sum of lengths of TextParts before the current one.
+    TextPart instances composed of multiple printable characters (those not
+    part of a terminal sequence) will be broken into multiple TextPart
+    instances, one per character.
+
+    This is useful for splitting text into its smallest indivisible TextPart
+    units: splitting strings into characters while not breaking up terminal
+    sequences.
+
+    :arg: parts: iterable of TextPart instances
+    :rtype: iterator of tuple pairs of (int, TextPart)
+
+    """
+    idx = 0
+    for part in parts:
+        if part.is_sequence:
+            yield idx, part
+            idx += len(part.ucs)
+        else:
+            for char in part.ucs:
+                yield idx, TextPart(char, False, None, None)
+                idx += 1
+
+
+def iter_parse(term, ucs):
+    r"""
+    Return an iterator of TextPart instances: terminal sequences or strings.
+
+    :arg ucs: str which may contain terminal sequences
+    :arg blessed.Terminal term: :class:`~.Terminal` instance.
+    :rtype: iterator of TextPart instances
+
+    TextPart is a :class:`collections.namedtuple` instance describing
+    either a terminal sequence or a series of printable characters.
+    Its parameters are:
+
+        - ``ucs``: str of terminal sequence or printable characters
+        - ``is_sequence``: bool for whether this is a terminal sequence
+        - ``name``: str of capability name or descriptive name of the terminal
+            sequence, or None if not a terminal sequence
```

Comment:
``or None if not`` => ``or empty string it not``

(let us also change the type to match). Consistent types, same meaning "Falsey"